app/WeatherApp.js

```diff
@@ -0,0 +1,178 @@
+import {
+  View,
+  StyleSheet,
+  SafeAreaView,
+  ImageBackground,
+  ScrollView,
+  ActivityIndicator,
+  Alert,
+} from "react-native";
+import React, { useEffect, useRef, useState } from "react";
+import * as NavigationBar from "expo-navigation-bar";
+import WeatherTabs from "../components/WeatherTabs";
+import AppHeader from "../components/AppHeader";
+import backgroundImg from "../assets/images/backgroundImg.jpeg";
+import { LinearGradient } from "expo-linear-gradient";
+import { useStore } from "../StoreProvider";
+
+const WeatherApp = () => {
+  const baseUrl = process.env.EXPO_PUBLIC_BASE_URL;
+  const {
+    getUserLocation,
+    location,
+    searchText,
+    weatherDetails,
+    setWeatherDetails,
+    setSearchText,
+    isLoading,
+    setIsLoading,
+    isAnimating,
+    setIsAnimating,
+    getRandomCityName,
+  } = useStore();
+
+  const timestamp = Date.now();
+  const { longitude, latitude } = location?.coords ?? {
+    longitude: "",
+    latitude: "",
+  };
+  const coordsURL = `${baseUrl}&q=${latitude},${longitude}&days=1&aqi=yes&alerts=no&timestamp=${timestamp}`;
```

Comment:
use new UrlSearchParams to create these, 
https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams

built in support for a log of operations. 

any other url with params in the app as well